code/modules/reagents/chemistry/reagents/trickwine_reagents.dm

```diff
@@ -0,0 +1,193 @@
+		if(iscarbon(M))
+			reac_volume = reac_volume / 4
+		M.electrocute_act(reac_volume, src, siemens_coeff = 1, flags = SHOCK_NOSTUN|SHOCK_TESLA)
+		do_sparks(5, FALSE, M)
+		playsound(M, 'sound/machines/defib_zap.ogg', 100, TRUE)
+
+/datum/reagent/consumable/ethanol/hearth_wine
+	name = "Hearthwine"
+	description = "A fiery brew utilized by members of the Saint-Roumain Militia, engineered to cauterize wounds in the field. Goes out in a blaze of glory when thrown."
+	color = "#ff8c00"
+	boozepwr = 70
+	taste_description = "the heat of battle"
+	glass_name = "Hearthwine"
+	glass_desc = "Fiery brew utilized by members of the Saint-Roumain Militia, engineered to cauterize wounds in the field. Goes out in a blaze of glory when thrown."
+	breakaway_flask_icon_state = "baflaskhearthwine"
+
+/datum/reagent/consumable/ethanol/hearth_wine/on_mob_life(mob/living/M)
+	M.adjust_bodytemperature(5 * TEMPERATURE_DAMAGE_COEFFICIENT, M.get_body_temp_normal())
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.bleed_rate = max(H.bleed_rate - 0.25, 0)
+	..()
+	. = 1
+
+/datum/reagent/consumable/ethanol/hearth_wine/expose_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH)
+		if(!iscarbon(M))
+			reac_volume = reac_volume * 2
+		M.fire_act()
+		var/turf/T = get_turf(M)
+		T.IgniteTurf(reac_volume)
+		new /obj/effect/hotspot(T)
+		T.hotspot_expose((reac_volume*10),(reac_volume*1))
+		var/turf/otherT
+		for(var/direction in GLOB.cardinals)
+			reac_volume = reac_volume / 10
+			otherT = get_step(T, direction)
+			otherT.IgniteTurf(reac_volume)
+			new /obj/effect/hotspot(otherT)
+			otherT.hotspot_expose((reac_volume*10),(reac_volume*1))
+
+/datum/reagent/consumable/ethanol/force_wine
+	name = "Forcewine"
+	description = "A fortifying brew utilized by members of the Saint-Roumain Militia, created to protect against the esoteric. Known to act defensively when thrown."
+	color = "#8b008b"
+	boozepwr = 70
+	taste_description = "the strength of your convictions"
+	glass_name = "Forcewine"
+	glass_desc = "A fortifying brew utilized by members of the Saint-Roumain Militia, created to protect against the esoteric. Known to act defensively when thrown."
+	breakaway_flask_icon_state = "baflaskforcewine"
+
+/datum/reagent/consumable/ethanol/force_wine/on_mob_metabolize(mob/living/M)
+	..()
+	ADD_TRAIT(M, TRAIT_ANTIMAGIC, "trickwine")
+	ADD_TRAIT(M, TRAIT_MINDSHIELD, "trickwine")
+	M.visible_message("<span class='warning'>[M] glows a dim grey aura</span>")
+
+/datum/reagent/consumable/ethanol/force_wine/on_mob_end_metabolize(mob/living/M)
+	M.visible_message("<span class='warning'>[M]'s aura fades away </span>")
+	REMOVE_TRAIT(M, TRAIT_ANTIMAGIC, "trickwine")
+	REMOVE_TRAIT(M, TRAIT_MINDSHIELD, "trickwine")
+	..()
+
+/datum/reagent/consumable/ethanol/force_wine/expose_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH)
+		var/turf/T = get_turf(M)
+		var/turf/otherT
+		reac_volume = reac_volume * 4
```

Comment:
You know, you can just set a new variable like "var/forcefield_dur = reac_volume * 4". Alternatively, you could just put "reac_volume*4" in the calls directly. Also, note that this will stack up a bunch of forcefields on the same tile if people are exposed rapidly. That's not a huge issue, since they'll decay on their own, but it might look strange.